052-cruise-control-api-users.md

```diff
@@ -0,0 +1,204 @@
+USER role: has access to all the GET endpoints except bootstrap and train.
+ADMIN role: has access to all endpoints.
+```
+
+Strimzi would then aggregate the Cruise Control API auth credentials into a centralized configuration, then store it in a secret where it would be mounted and used by Cruise Control.
+
+### How API user management is handled elsewhere
+
+Cruise Control is not the only Kafka component which has API users.
+For consitency across the code base, we should also consider the following components when designing our solution:
+
+* Kafka users
+* KafkaConnect users
+
+#### Kafka users
+
+In Strimzi, Kafka users are managed via custom resources.
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaUser
+metadata:
+  name: my-user-1
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: tls
+  authorization:
+    type: simple
+    acls:
+      ...
+```
+These resources are watched and managed by the the Strimzi User Operator, which creates Kafka users based on the descriptions provided by the resources.
+
+For more details, checkout the Strimzi documentation [here](https://strimzi.io/docs/operators/latest/deploying#proc-configuring-kafka-user-str)
+
+#### KafkaConnect users
+
+At this time, Strimzi doesn't offer any method of creating or configuring KafkaConnect API users.
+That being said, the direction we take with creating and managing Cruise Control API users could be used as a precedent for its design in the future.
+
+## Alternatives
+
+### Cruise Control user custom resource
+
+Following the pattern used for Kafka users, we could have `CruiseControlUser` resources.
+
+This would make the management of Cruise Control API users consitent with how we manage Kafka users
+
+The spec could look somthing like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: CruiseControlUser
+metadata:
+  name: my-user-1
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  role: USER (3)
+  password:
+    valueFrom:
+      secretKeyRef:
+        name: my-secret (1)
+        key: my-password (2)
+```
+
+(1) The name of the secret containing the predefined password.
+
+(2) The key for the password stored inside the secret.
+
+(3) By default Cruise Control defines three roles: `VIEWER`, `USER` and `ADMIN`. 
+For more details checkout the [Cruise Control Wiki](https://github.com/linkedin/cruise-control/wiki/Security#authorization)
+```
+VIEWER role: has access to the most lightweight kafka_cluster_state, user_tasks and review_board endpoints.
+USER role: has access to all the GET endpoints except bootstrap and train.
+ADMIN role: has access to all endpoints.
+```
+
+The problem with this approach is that it would add more complexity to the code base since it would require:
+* A new Strimzi `CruiseControlUser` custom resource
+* A watch on the new `CruiseControlUser` resource
+* A reconciler to manage the `CruiseControlUser` resource.
+
+### Developer managed API user configuration
```

Comment:
This is probably something I personally would prefer.

Reply:
I have read all the feedback and thought about this a lot and I understand your reasoning and agree that this implementation would make more sense given the circumstances. I have just updated the proposal to highlight this a the suggested approach.